src/charm.py

```diff
@@ -121,6 +125,18 @@ def _is_legacy_v1_relation(self, relation):
 
         return True
 
+    def _configure_ingress(self, _) -> None:
+        """Make sure the traefik route and tracing relation data are up to date."""
+        if not self.unit.is_leader():
+            return
+
+        if self._ingress.is_ready():
+            self._ingress.submit_to_traefik(
+                self._ingress_config, static=self._static_ingress_config
+            )
+            self._update_tracing_v1_relations()
```

Comment:
you could do this conditionally on there being an ingress url already. Pointless to update relations if traefik hasn't sent you an url back yet. We should only do this on ingress ready in fact
`if self._ingress.url: ...`